### PR TITLE
Bitstamp: Add missing API fields

### PR DESCRIPTION
**Pull request by Beadko**
# PR Description

Issues picked up during PR#1423 but were considered out of scope. In addition, the main error checks changed from assert to require.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [x] golangci-lint run
- [x] Test Bitstamp_test

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/bitstamp/bitstamp_test.go [hunk 1]
```diff
@@ -354,15 +353,19 @@ func TestGetOpenOrders(t *testing.T) {
...
-			assert.Equal(t, 0, res.Type, "type should match")
+			assert.Equal(t, "2022-01-31 14:43:15", res.DateTime, "DateTime should match")
+			assert.Equal(t, int64(1234123412341234), res.ID, "ID should match")
+			assert.Equal(t, 0.50000000, res.Amount, "Amount should match")
+			assert.Equal(t, 100.00, res.Price, "Price should match")
+			assert.Equal(t, 0, res.Type, "Type should match")
+			assert.Equal(t, 0.50000000, res.AmountAtCreate, "AmountAtCreate should match")
+			assert.Equal(t, 110.00, res.LimitPrice, "LimitPrice should match")
+			assert.Equal(t, "1234123412341234", res.ClientOrderID, "ClientOrderID should match")
+			assert.Equal(t, "BTC/USD", res.Market, "Market should match")
```
Comment (Beadko) [inline exchanges/bitstamp/bitstamp_test.go:368]: One comment was regarding the Currency is not set for []Order type. Bitstamp open orders for markets do not include currency in the response so I have added "Market" which was missing

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Nice work Beadko!

File: exchanges/bitstamp/bitstamp_test.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func TestGetFee(t *testing.T) {
 
 	// CryptocurrencyTradeFee Basic
 	fee, err := b.GetFee(context.Background(), feeBuilder)
-	assert.NoError(t, err, "GetFee should not error")
```
[the remark this answers is not in the document]
Comment (Beadko) [reply exchanges/bitstamp/bitstamp_test.go:67]: Added

File: exchanges/bitstamp/bitstamp_test.go [hunk 3]
```diff
@@ -64,7 +64,10 @@ func TestGetFee(t *testing.T) {
...
+	if !mockTests {
+		sharedtestvalues.SkipTestIfCredentialsUnset(t, b)
+	}
```
Comment (gloriousCode) [inline exchanges/bitstamp/bitstamp_test.go:69]:
```suggestion
	if !mockTests {
    		sharedtestvalues.SkipTestIfCredentialsUnset(t, b)
	}
	fee, err := b.GetFee(context.Background(), feeBuilder)
```

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Just one really minor thing 😄

File: exchanges/bitstamp/bitstamp_test.go (hunk 3 above)
Comment (Beadko) [reply exchanges/bitstamp/bitstamp_test.go]: done :)

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks beadko!

File: exchanges/bitstamp/bitstamp_types.go [hunk 4]
```diff
@@ -114,28 +116,35 @@ type UserTransactions struct {
...
-		TradeID int64   `json:"tid"`
-		USD     float64 `json:"usd,string"`
-		Price   float64 `json:"price,string"`
-		Fee     float64 `json:"fee,string"`
-		BTC     float64 `json:"btc,string"`
+	AmountRemaining float64 `json:"amount_remaining,string"`
+	Type            int     `json:"type"`
+	ID              string  `json:"id"`
+	DateTime        string  `json:"datetime"`
+	Status          string
```
Comment (shazbert) [inline exchanges/bitstamp/bitstamp_types.go:137]: missing json tag

**Review by shazbert — CHANGES_REQUESTED** [review]
One thing from me everything else looks good. Nice work.

File: exchanges/bitstamp/bitstamp_types.go (hunk 4 above)
Comment (Beadko) [reply exchanges/bitstamp/bitstamp_types.go]: oops,  added now.

**Review by shazbert — APPROVED** [review]
Thanks Bea! changes tack.

**Review by thrasher- — APPROVED** [review]
tACK mock tests on and off (unauth). Thanks @Beadko !

**Pull request merged**